File: packages/core/src/types/business.ts
export enum BusinessStatus {
  Draft         = 'draft',
  PendingReview = 'pending_review',
  Pending       = 'pending',
  Active        = 'active',
  Inactive      = 'inactive',
}

export enum BusinessType {
  Heladeria   = 'heladeria',
  Barberia    = 'barberia',
  Estetica    = 'estetica',
  Restaurante = 'restaurante',
  Panaderia   = 'panaderia',
  Gym         = 'gym',
  Mecanico    = 'mecanico',
  Otro        = 'otro',
}

/** Legal document kinds a seller can publish on their storefront. */
export enum LegalDocType {
  Terms   = 'terms',
  Privacy = 'privacy',
}

/** A single legal document. `content` is sanitized HTML produced by the rich-text editor. */
export interface LegalDocument {
  content: string
  updatedAt: string | null
  /** Set when the content was last produced by AI generation; null/undefined if hand-written. */
  generatedAt?: string | null
}

/** Per-business legal documents, keyed by {@link LegalDocType}. */
export type LegalDocs = Partial<Record<LegalDocType, LegalDocument>>

export interface BusinessTheme {
  primary: string
  secondary: string
  accent: string
  background: string
  font: string
  emoji: string
  gradient: [string, string]
}

export interface Business {
  id: string
  slug: string
  name: string
  type: BusinessType
  whatsapp: string
  city: string
  state?: string
  contactName?: string
  logo?: string
  tagline?: string
  legalDocs?: LegalDocs
  theme: BusinessTheme
  status: BusinessStatus
  plan: 'free' | 'pro' | 'growth'
  ownerId?: string
  domain?: string
  ownerEmail?: string
  submittedAt?: string
  ownerApprovedAt?: string
  demoGeneratedAt?: string
  sentAt?: string
  acceptedAt?: string
  activatedAt?: string
  createdAt: string
  updatedAt: string
}

File: apps/catalog-api/app/domain/business.py
"""Business domain — pure Python, no FastAPI, no Firestore."""
import re
from enum import Enum


class BusinessStatus(str, Enum):
    PENDING  = "pending"
    ACTIVE   = "active"
    INACTIVE = "inactive"


class BusinessType(str, Enum):
    HELADERIA   = "heladeria"
    BARBERIA    = "barberia"
    ESTETICA    = "estetica"
    RESTAURANTE = "restaurante"
    PANADERIA   = "panaderia"
    GYM         = "gym"
    MECANICO    = "mecanico"
    OTRO        = "otro"


VALID_TRANSITIONS: dict[BusinessStatus, list[BusinessStatus]] = {
    BusinessStatus.PENDING:  [BusinessStatus.ACTIVE],
    BusinessStatus.ACTIVE:   [BusinessStatus.INACTIVE],
    BusinessStatus.INACTIVE: [BusinessStatus.ACTIVE],
}

ACTION_TO_STATUS: dict[str, BusinessStatus] = {
    "activate":   BusinessStatus.ACTIVE,
    "deactivate": BusinessStatus.INACTIVE,
    "reactivate": BusinessStatus.ACTIVE,
}

OWNER_PATCH_FIELDS = frozenset({"tagline", "theme", "name", "whatsapp"})


def resolve_new_status(current: BusinessStatus, action: str) -> BusinessStatus:
    """Return the target status for an action, or raise ValueError."""
    new_status = ACTION_TO_STATUS.get(action)
    if not new_status:
        raise ValueError(f"Unknown action '{action}'")
    allowed = VALID_TRANSITIONS.get(current, [])
    if new_status not in allowed:
        raise ValueError(f"Cannot transition from '{current}' to '{new_status}'")
    return new_status


def slugify(text: str) -> str:
    t = text.lower()
    for a, b in [("á","a"),("é","e"),("í","i"),("ó","o"),("ú","u"),("ñ","n"),("ü","u")]:
        t = t.replace(a, b)
    t = re.sub(r"[^a-z0-9\s-]", "", t)
    return re.sub(r"\s+", "-", t.strip())[:40]
